aiohttp/client.py

```diff
@@ -309,12 +352,13 @@ def request(self, method, url, **kwargs):
                         expect100=expect100, loop=self._loop,
                         response_class=self._response_class,
                         proxy=proxy, proxy_auth=proxy_auth, timer=timer,
-                        session=self, auto_decompress=self._auto_decompress,
+                        session=self,
                         ssl=ssl, proxy_headers=proxy_headers, traces=traces)
 
                     # connection timeout
                     try:
-                        with CeilTimeout(self._conn_timeout, loop=self._loop):
+                        with CeilTimeout(self._timeout.connect,
```

Comment:
I think this still means that the connect timeout includes the time waiting for a connector from the pool no?

Reply:
Nice catch!
Let's keep `ClientTimeout.connect` as is for backward compatibility but implement  `.sock_connect` for actual connection establishment time.